src/main/java/com/google/sps/data/Comment.java

```diff
@@ -1,11 +1,48 @@
 package com.google.sps.data;
 
-public class Comment{
+public final class Comment{
     private final String name;
     private final String comment;
+    private final long id;
 
-    public Comment(String name, String comment){
+    public Comment(String name, String comment, long id){
         this.name = name;
         this.comment = comment;
+        this.id = id; 
+    }
+
+    public String getDisplayName(){
+        return name; 
+    }
+
+    public String getComment(){
+        return comment; 
+    }
+
+    public long getID(){
+        return id; 
+    }
+
+    @Override
+    public boolean equals(Object o) {   
+        if (!(o instanceof Comment)) { 
+            return false; 
+        } 
+        Comment that = (Comment) o;
+        return that.getComment().equals(this.comment)
+            && that.getDisplayName().equals(this.name);
+    } 
+
+    @Override
+    public int hashCode(){
+        //TODO
```

Comment:
this will have to return an int for it to compile...

you can return 42; for now. 

TODO's should be of the form 

TODO(username): explanation of the todo (such as implement hashCode().